### PR TITLE
Cmake/build split

### PR DESCRIPTION
**Pull request by AlexBuccheri**
This PR splits up the src of minimax, such that modules and routines common to all GX libs are now in their own lib: GX-common. I envision if one builds for abinit, for example, it will compile GX-common and the libs specific to abinit. Same for FHI, same for exciting. 

It also cleans up the minimax src and adds build flags for Intel and GCC. I realise the PR does multiple things, but I think that's ok as we quickly flesh out the structure. 

I note that the structure of the build and install directories requires some further cleaning. 

@jmbeuken I don't expect to change the src structure now
@gmatteo Can you comment on the flags? Also, I've arbitrarily set compiler support to GCC7. Intel needs deciding. 

TODOs
* Comment on and set compiler support

### CODE REVIEW

File: GX-common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# CMakeLists.txt for GreenX Common Modules, required by all other libs in
+# the package
+
+add_library(GXCommon "")
+
+set_target_properties(GXCommon
+        PROPERTIES
+        VERSION 0.0.1
+        SOVERSION 0.01
```
Comment (AlexBuccheri) [inline GX-common/CMakeLists.txt:9]: - [ ] 0.0.1

**Comment by AlexBuccheri** [comment]
Build fails due to the stricter compiler flags

File: GX-common/CMakeLists.txt (hunk 1 above)
Comment (AlexBuccheri) [reply GX-common/CMakeLists.txt:9]: @jmbeuken Could you please not mark things as resolved unless the changes have been pushed, else it's easy to miss things. I have now made the change

**Review by aziziph — APPROVED** [review]
Good idea. Sounds OK!

**Pull request merged**